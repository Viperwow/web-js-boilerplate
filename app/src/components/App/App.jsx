// Vendors
import { hot } from 'react-hot-loader'; // Is being used to save React components state after HRM
import React, { Component } from 'react';
// Components
import Layout from 'src/components/Layout';

const NO_DATA_MESSAGE = 'Put_content_here';

class App extends Component {
  render() {
    return (
      <div className="root">
        <Layout>
          { NO_DATA_MESSAGE }
        </Layout>
      </div>
    );
  }
}

export default hot(module)(App);
