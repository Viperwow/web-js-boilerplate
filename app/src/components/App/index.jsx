// Vendors
import { hot } from 'react-hot-loader'; // Is being used to save React components state after HRM
import React, { Component } from 'react';
// Components
import Layout from 'src/components/Layout';
import Example from 'src/controllers/Example';
import Example2 from 'src/controllers/Example2';

const NO_DATA_MESSAGE = 'Put_content_here';

class App extends Component {
  render() {
    return (
      <div className="ny-app">
        <Layout>
          { NO_DATA_MESSAGE }
          { Example.mult(2, 2) }
          { Example2.mult(3, 3) }
        </Layout>
      </div>
    );
  }
}

export default hot(module)(App);
