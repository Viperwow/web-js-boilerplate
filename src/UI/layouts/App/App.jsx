import React from 'react';
import {
  Route,
  Router,
  Switch,
} from 'react-router-dom';
import {getHistory} from 'src/helpers/history';
import Layout from 'src/UI/layouts/Layout';

const App = () => (
  <Router history={getHistory()}>
    <Switch>
      <Route path="/" render={Layout} />
    </Switch>
  </Router>
);

export default App;
