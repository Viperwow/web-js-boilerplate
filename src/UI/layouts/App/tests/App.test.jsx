import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import App from '../App';

jest.unmock('../App');

describe('App component should render correctly', () => {
  test('With default props', () => {
    const output = shallow(<App />);

    expect(shallowToJson(output))
      .toMatchSnapshot();
  });
});
