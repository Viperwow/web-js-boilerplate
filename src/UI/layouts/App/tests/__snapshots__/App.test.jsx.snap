// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly With default props 1`] = `
<Router
  history={Object {}}
>
  <Switch>
    <Route
      path="/"
      render={[MockFunction undefined]}
    />
  </Switch>
</Router>
`;
